gcode_metadata/metadata.py

```diff
@@ -99,6 +99,53 @@ def same_or_nothing(value_list):
     return value_list[0]
 
 
+def extract_data(input_string):
+    """Extracts metadata from the filename
+        >>> extract_data("MP_PLA,PLA_MK3SMMU3_3h22m.gcode")
+        {'name': 'MP_', 'nozzle': None, 'height': None, 'material': 'PLA', 'printer': 'MK3SMMU3', 'time': '3h22m'}
+        >>> extract_data("sh_bn_0.6n_0.32mm_PETG_MK4_8h55m.gcode")
```

Comment:
try to `...` in doctest, and see to python doc for more tips.
By the way `extract_data("MP_PLA,PLA_MK3SMMU3_3h22m.gcode")['printer']` is one of way to check one attribute in returned dictionary.

Reply:
+ Add a "garbage" name gcode, eg. a.gcode, only days etc.